[Feature] Update Link active state

narin opened the pull request:
<!-- PR title naming convention:
'[Issue type] Brief summary of issue suitable for changelog or copy/paste issue title',
where Issue type = bug, feature, spike, CU (code upkeep), etc.-->

<!-- Preferred branch naming convention:
'[Issue type]/[Issue #]-[Your name]-[Summary of issue]',
where Issue type = bug, feature, spike, CU (code upkeep), etc.-->

## Description of Change
<!-- Describe the change and context with which it was made beyond ACs unless straightforward.
Consider:
 - What is relevant to code reviewer(s) and not in the ticket?
 - What context may be relevant to a future dev or you in 6 months about this PR?
 - Did the course of work lead to notable dead ends? If so, why didn't they pan out?
 - Did the change add new dependencies? Why?
 - Were there important sources to link? Examples: an open bug with a dependency project, an article of someone else solving the same problem that was partially or wholly copied, external documentation relevant to solution
 -->
Replaces bold styling with background color for Link active state as per [figma design](https://www.figma.com/proto/Zzt8z60hCtdEzXx2GFWghH/branch/HRsHkww7nAqSdquMQ7uEZ4/%F0%9F%93%90-Component-Library?page-id=224%3A314&type=design&node-id=2186-478&viewport=1743%2C-866%2C0.74&t=lYrOBHRm2XOOBM8t-1&scaling=min-zoom). For light mode background is `grayDark` and dark mode it is `grayLighter`. Also added unit tests for the pressed state styling.


#### Testing Packages
<!-- List or range of alpha/beta packages published in association with this PR, if any -->
N/A

### Screenshots/Video
<!-- Add screenshots or video as needed; before/after recommended if appropriate. 
Convenience side-by-side formatting:
Before/after: <img src="" width="49%" />&nbsp;&nbsp;<img src="" width="49%" />
Accordion before/after: <details><summary>Before/after</summary><img src="" width="49%" />&nbsp;&nbsp;<img src="" width="49%" /></details>
-->
#### iOS

https://github.com/department-of-veterans-affairs/va-mobile-library/assets/786704/d5b0a885-8d23-4d61-a063-a9a150ba4b19


### Android

https://github.com/department-of-veterans-affairs/va-mobile-library/assets/786704/a5b0bc26-945f-450d-b4d1-3bae6df10db6



#### Web

https://github.com/department-of-veterans-affairs/va-mobile-library/assets/786704/50287472-5486-4843-810c-126f4e31a848


## Testing
<!-- Describe testing conducted to validate changes.
Consider highlighting:
- What testing was not explicitly done and may be relevant for QA? 
- Edge cases validated
- Special situations that could not be tested
- Any testing performed in a consuming app -->

- [x] Tested on iOS <!-- simulator is fine -->
- [x] Tested on Android <!-- simulator is fine -->
- [x] Tested on Web

## PR Checklist
Code reviewer validation:
- General
	- [x] PR is linked to ticket(s)
	- [x] PR has `changelog` label applied if it's to be included in the changelog
	- [x] Acceptance criteria: 
		- All satisfied _or_
		- Documented reason for not being performed _or_
		- Split to separate ticket and ticket is linked by relevant AC(s)
	- [x] Above PR sections adequately filled out
	- [x] If any breaking changes, in accordance with the [pre-1.0.0 versioning guidelines](https://github.com/department-of-veterans-affairs/va-mobile-library#versioning-policy): a CU ticket has been created for the VA Mobile App detailing necessary adjustments with the package version that will be published by this ticket
- Code
	- [x] Tests are included if appropriate (or split to separate ticket)
	- [x] New functions have proper TSDoc annotations

## Publish
<!-- Most changes entail a version increment; section can be removed for PRs exclusively within non-ship-relevant files (e.g. unit tests, Storybook stories) -->
If changes warrant a new version [per the versioning guidelines](https://github.com/department-of-veterans-affairs/va-mobile-library#versioning-policy) and the PR is approved and ready to merge:
- [x] Merge `main` into branch
- [ ] Merge branch to `main`
- [ ] Verify that [Check Component Integrations](https://github.com/department-of-veterans-affairs/va-mobile-library/actions/workflows/check-component-integrations.yml) workflow ran successfully
- [ ] [Publish new version](https://github.com/department-of-veterans-affairs/va-mobile-library/actions/workflows/publish.yml) after merging

TimRoe commented on packages/components/src/components/Link/Link.tsx:
```diff
@@ -271,37 +274,35 @@ export const Link: FC<LinkProps> = ({
     <Pressable {...pressableProps} testID={testID}>
       {({ pressed }: PressableStateCallbackType) => (
-        <>
+        <View style={getViewStyle(pressed)}>
```
Fairly inconsequential, but pretty sure this could also just be done against the Pressable itself.

TimRoe replied:
Actually, turns out this is consequential--poking around on a "what if" the current state breaks the centering of the text alignment if you make the icon bigger (e.g. width 50, height 50). On main branch the link text remains centered, on your branch it aligns near the top.

So either need to also add alignItems center to defaultStyle or rework slightly to do do the pressed styling on Pressable. 

Kind of on the fence: on the Pressable seems "cleaner" in my mind, but maybe moving the style down to the View (and removing from Pressable) is more defensive coding lest future changes add a View here and break the inheritance of style from Pressable similarly.

TimRoe reviewed (changes requested):
Mostly good, but looks like it did cause a minor bug with vertical centering within the component--see other comments for thoughts on resolution.

narin commented:
Moved pressed styling to Pressable which fixed the vertical alignment and simplified things.

![image](https://github.com/department-of-veterans-affairs/va-mobile-library/assets/786704/a88f2a1c-5249-4df0-8fd5-3d4093bdaf5a)

TimRoe commented on packages/components/src/components/Link/Link.tsx:
```diff
@@ -241,7 +243,7 @@ export const Link: FC<LinkProps> = ({
     _icon === 'no icon' ? null : (
       <View style={iconViewStyle}>
-        <Icon fill={linkColor} {..._icon} />
+        <Icon fill={linkColor} {..._icon} width={50} height={50} />
```
Remove testing code.
```suggestion
        <Icon fill={linkColor} {..._icon} />
```
